omod/src/main/java/org/openmrs/module/owa/web/controller/OwaManageController.java

```diff
@@ -102,7 +102,7 @@ public boolean settingsValid() {
 
 	@ModelAttribute("appBaseUrl")
 	public String getAppBaseUrl() {
-		return Context.getAdministrationService().getGlobalProperty(AppManager.KEY_APP_BASE_URL);
+		return appBaseUrl;
```

Comment:
I'm surprised it works. If someone changes the GP value, even though it is saved, the cached appBaseUrl is still displayed to the user on the manage page. I'd not cache the value.